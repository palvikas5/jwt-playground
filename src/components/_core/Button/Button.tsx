import React from 'react';
import { noop } from '../../../utils/noop';
import { StyledButton } from './button.style';

interface ButtonProps {
  onClick?: () => void;
  children: string | number;
}

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  ({ onClick, children }: ButtonProps, ref) => {
    return (
      <StyledButton type='button' onClick={onClick} ref={ref}>
        {children}
      </StyledButton>
    );
  },
);

Button.defaultProps = {
  onClick: noop,
};

export default Button;
