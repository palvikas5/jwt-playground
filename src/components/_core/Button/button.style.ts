import styled from 'styled-components';

const StyledButton = styled.button`
  display: block;
  color: white;
  background-color: #3e8efa;
  text-transform: uppercase;
  padding: 12px 15px;
  border: none;
  border-radius: 3px;
`;

export { StyledButton };
