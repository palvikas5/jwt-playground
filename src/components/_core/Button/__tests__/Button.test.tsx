import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Button from '../Button';

describe('<Button />', () => {
  it('should render successfully', () => {
    expect.assertions(1);

    const { container } = render(
      <Button onClick={jest.fn()}>button-text</Button>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should contains allowed values as children', () => {
    expect.assertions(2);

    const wrapper = render(<Button>button-text</Button>);
    expect(wrapper.getByText(/^button-text$/)).toBeInTheDocument();

    const wrapper1 = render(<Button>{1}</Button>);
    expect(wrapper1.getByText(/^1$/)).toBeInTheDocument();
  });

  it('should trigger onClick function', () => {
    expect.assertions(1);

    const clickHandler = jest.fn();
    const { getByText } = render(
      <Button onClick={clickHandler}>button-text</Button>,
    );
    const button = getByText('button-text');
    fireEvent.click(button);
    expect(clickHandler).toHaveBeenCalledTimes(1);
  });
});
