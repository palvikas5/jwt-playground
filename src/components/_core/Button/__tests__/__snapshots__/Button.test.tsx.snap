// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render successfully 1`] = `
.c0 {
  display: block;
  color: white;
  background-color: #3e8efa;
  text-transform: uppercase;
  padding: 12px 15px;
  border: none;
  border-radius: 3px;
}

<div>
  <button
    class="c0"
    type="button"
  >
    button-text
  </button>
</div>
`;
